src/XR/features/WebXRNearInteraction.ts

```diff
@@ -321,148 +361,172 @@ export class WebXRNearInteraction extends WebXRAbstractFeature {
                 }
                 return pick;
             };
-            let populateNearInteractionInfo = (nearInteractionInfo: Nullable<PickingInfo>): Nullable<PickingInfo> => {
-                let result = null;
+            let populateNearInteractionInfo = (nearInteractionInfo: Nullable<PickingInfo>): PickingInfo => {
+                let result = new PickingInfo();
+
                 let nearInteractionAtOrigin = false;
                 let nearInteraction = nearInteractionInfo && nearInteractionInfo.pickedPoint && nearInteractionInfo.hit;
                 if (nearInteractionInfo?.pickedPoint) {
                     nearInteractionAtOrigin = nearInteractionInfo.pickedPoint.x === 0 && nearInteractionInfo.pickedPoint.y === 0 && nearInteractionInfo.pickedPoint.z === 0;
                 }
                 if (nearInteraction && !nearInteractionAtOrigin) {
-                    result = nearInteractionInfo;
-                    controllerData.nearInteraction = true;
+                    result = nearInteractionInfo!;

```

Comment:
Does the conditional check above not guarantee to the typescript compiler that nearInteractionInfo exists?

Reply:
yep it should, but it doesn't, compiler requires the "!" sign. I think the compiler still doesn't decompose these kinds of booleans.